packages/grid/data-grid/src/tests/pagination.DataGrid.test.tsx

```diff
@@ -187,5 +188,43 @@ describe('<DataGrid /> - Pagination', () => {
       fireEvent.click(screen.getByRole('button', { name: /next page/i }));
       expect(getColumnValues()).to.deep.equal(['Nike 1']);
     });
+
+    describe('AutoPageSize', () => {
+      it('should always render the same amount of rows and fit the viewport', () => {
+        const TestCaseAutoPageSize = (
+          props: Partial<DataGridProps> & { nbRows?: number; nbCols?: number; height?: number },
+        ) => {
+          const data = useData(props.nbRows || 100, props.nbCols || 10);
+
+          return (
+            <div style={{ width: 300, height: props.height || 300 }}>
+              <DataGrid
+                columns={data.columns}
+                rows={data.rows}
+                autoPageSize
+                pagination
+                {...props}
+              />
+            </div>
+          );
+        };
+        render(<TestCaseAutoPageSize nbRows={27} height={780} />);
+        let rows = document.querySelectorAll('.MuiDataGrid-viewport [role="row"]');
+        expect(rows.length).to.equal(12);
```

Comment:
Can we change this number to be calculated? Like it is done for the density tests?

Maybe `GRID_HEIGHT / ROW_HEIGHT`? Also, how do we know that there are only 3 pages? I guess because of the height of the grid so, `NUMBER_OF_PAGES = GRID_HEIGHT / (ROW_HEIGHT * TOTAL_NUMBER_OF_ROWS)`?